analyzer/codechecker_analyzer/analyzers/result_handler_base.py

```diff
@@ -92,8 +93,36 @@ def analyzer_action_str(self):
             os.path.join(self.buildaction.directory,
                          self.analyzed_source_file))
 
-        build_info = source_file + '_' + \
-            self.buildaction.original_command
+        # In case of "make 4.3" depending on compile-time options "make" tool
+        # can be built so a subprocess named cc1build will be logged by
+        # "CodeChecker log".
+        # See posix_spawn() option:
+        # https://lists.gnu.org/archive/html/info-gnu/2020-01/msg00004.html
+        # In this case the -o output argument of cc1build command is a randomly
+        # named temporary file. We can't afford dynamic parts in the original
+        # build command, because its hash is used for identification in the
+        # plist file name.
+        #
+        # TODO: This solution is considered a workaround, because the real
+        # question is why such a subprocess is logged? Can cc1build be always
+        # used as a traditional g++ command?
+        #
+        # Note that some information loss occurs during the following algorithm
+        # because ' '.join(shlex.split(cmd)) is not necessarily equal to cmd:
+        # g++ -DVAR="hello world" main.cpp
+
+        args = shlex.split(self.buildaction.original_command)
```

Comment:
what do you think about storing the `self.buildaction.original_command` as a list instead of a string so this split only needs to be done once and will not be done differently at multiple places?

Reply:
Well, we can try this approach. We'll have to be very careful though, because the type of this attribute would change and we'll have to check if it's used correctly everywhere. By the way the joint version is used at some places, so in case of storing a list, sometimes we'll need to join it. Maybe joining is the less frequent operation. However, we'll have to use `shlex.join()` instead of string's `join()` because this latter one comes with information loss and results some bugs. Unfortunately `shlex.join()` is introduced in one of the latest Python versions. The default Python version on my Ubuntu 18.04 machine is 3.6.9 which doesn't support `shlex.join()`. Can we expect a newer Python version from the users?

Reply:
Let's keep it as is for now.